proposals/lip-0045.md

```diff
@@ -1411,15 +1411,41 @@ def getTerminatedOutboxAccount(chainID: ChainID) -> TerminatedOutboxAccount:
     return terminatedOutboxAccount(chainID)
 ```
 
+### Endpoints for Off-Chain Services
+
 #### isChainIDAvailable
 
 ```python
 def isChainIDAvailable(chainID: ChainID) -> bool:
-    if chainAccount(chainID) does not exist:
-        return True
-    return False
+    if chainAccount(chainID) exist:
+        return False
+    
+    if chainAccount.chainID[0] != getMainchainID()[0]:
+       return False
+       
+    if chainAccount.chainID == getMainchainID():
+       return False
+           
+    return True
+```
+
+#### isChainNameAvailable
+
+```python
+def isChainIDAvailable(chainName: str) -> bool:
+    if if not re.match(r"^[a-z0-9!@$&_.]+$", chainName):
+        return False
+    
+    if len("chainName") < minLength or len("chainName") > maxLength:
```

Comment:
```suggestion
    if len(chainName) < MIN_CHAIN_NAME_LENGTH or len(chainName) > MAX_CHAIN_NAME_LENGTH:
```

Reply:
`chainName` is a variable, not a string. Also, we should use the already defined constant names.

Reply:
Done